File: atividades/atividade13/ComparaString.py


import sys


string_1 = sys.argv[1]
string_2 = sys.argv[2]

print(f"{string_1}: {len(string_1)} ")
print(f"{string_2}: {len(string_2)} ")

if string_1.find(string_2) == 0:
    print(f"{string_1} contém {string_2}")

else:
    print(f"{string_1} não contém {string_2}")


if string_2.find(string_1) == 0:
    print(f"{string_2} contém {string_1}")

else:
    print(f"{string_2} não contém {string_1}")


print(string_1[::-1])
print(string_2[::-1])

File: atividades/atividade13/ComparaString.c
#include<stdio.h>
#include<string.h>

int main (int argc, char *argv[]){
	char * string_1 = argv[1];
	char * string_2 = argv[2];

	printf("%s: %lu\n", string_1, strlen(string_1));
	printf("%s: %lu\n", string_2, strlen(string_2));

	char * pch = strstr(string_1, string_2);

	if ( pch != NULL){
		printf("%s contém %s\n", string_1, string_2);
	}
	else{
		printf("%s não contém %s\n", string_1, string_2);
	}

	pch = strstr(string_2, string_1);

        if ( pch != NULL){
                printf("%s contém %s\n", string_2, string_1);
        }
        else{
                printf("%s não contém %s\n", string_2, string_1);
        }



	char caracter;
	int j = 0;
	int len = strlen(string_1) - 1;
	for (int i = len; i > len/2; i--, j++){
		caracter= string_1[j];
		string_1[j] = string_1[i];
		string_1[i] = caracter;
	}

	printf("%s\n", string_1);


        caracter;
        j = 0;
        len = strlen(string_2) - 1;
        for (int i = len; i > len/2; i--, j++){
                caracter= string_2[j];
                string_2[j] = string_2[i];
                string_2[i] = caracter;
        }

        printf("%s\n", string_2);


}
